core/src/peer_id.rs

```diff
@@ -28,12 +30,15 @@ use thiserror::Error;
 /// automatically used as the peer id using an identity multihash.
 const MAX_INLINE_KEY_LENGTH: usize = 42;
 
+/// CIDv1 MUST use the libp2p key codec to identify the content of the CID.
+const LIBP2P_KEY_CODEC: u64 = 0x72;
+
 /// Identifier of a peer of the network.
 ///
-/// The data is a multihash of the public key of the peer.
+/// The data is a cid of the public key of the peer.
 #[derive(Clone, Copy, Eq, Hash, Ord, PartialEq, PartialOrd)]
```

Comment:
If I am not mistaken the implementation of `Eq` is simply derived on both `PeerId` and `Cid`. Given that we store all `PeerID`s as CIDv1s now, an old style `PeerId` and a new style `PeerId` both parsed from a `String` would still equal, correct? Could you add a test to make sure this is the case going forward?